lib/reindex/s3_bucketd.py

```diff
@@ -266,23 +266,15 @@ def index_bucket(client, bucket):
     '''
     bucket_total = client.count_bucket_contents(bucket)
     mpus = client.list_mpus(bucket)
-    shadowbucket = bucket._replace(name=MPU_SHADOW_BUCKET_PREFIX + bucket.name)
     if not mpus:
-        mpu_total = BucketContents(shadowbucket, 0, 0)
-    else:
-        mpu_totals = [client.count_mpu_parts(m) for m in mpus]
-        mpu_part_count = 0
-        mpu_total_size = 0
-        for mpu in mpu_totals:
-            mpu_part_count += mpu.obj_count
-            mpu_total_size += mpu.total_size
-        mpu_total = BucketContents(
-                shadowbucket,
-                mpu_part_count,
-                mpu_total_size
-            )
-    return bucket_total, mpu_total
+        return bucket_total
 
+    total_size = bucket_total.total_size
+    mpu_totals = [client.count_mpu_parts(m) for m in mpus]
+    for mpu in mpu_totals:
+        total_size += mpu.total_size
```

Comment:
could use sum here

90% sure sum takes comprehensions... 
```suggestion
    total_size += sum(client.count_mpu_parts(m) for m in mpus)
```

Reply:
Since I'm referencing `.total_size` I'd need a something like
```python
total_size += sum(t.total_size for t in (client.count_mpu_parts(m) for m in mpus))
```
or split in two

```python
mpu_totals = [client.count_mpu_parts(m) for m in mpus]
total_size += sum(m.total_size for m in mpu_totals)
```
I'm not really sure either is any better

Reply:
ooh yea, missed that part. Yea kinda much for a single line.